src/Pact/Compile.hs

```diff
@@ -196,24 +294,27 @@ literal = lit >>= \LiteralExp{..} ->
 
 deftable :: Compile (Term Name)
 deftable = do
-  (mn,mh) <- currentModule
+  ModuleState{..} <- moduleState
   AtomExp{..} <- userAtom
   ty <- optional (typed >>= \t -> case t of
                      TyUser {} -> return t
                      _ -> expected "user type")
   m <- meta ModelNotAllowed
-  TTable (TableName _atomAtom) mn mh
+  TTable (TableName _atomAtom) _msName _msHash
     (fromMaybe TyAny ty) m <$> contextInfo
 
 
-bless :: Compile (Term Name)
-bless = TBless <$> hash' <*> contextInfo
+bless :: Compile ()
+bless = do
+  h <- hash'
+  overModuleState msBlessed (h:)
```

Comment:
There's got to be some killer lens one-liners that do all of this for us. I'm gonna go hunt and see